opm/simulators/utils/HDF5File.cpp

```diff
@@ -123,4 +142,23 @@ void HDF5File::read(const std::string& group,
     H5Dclose(dataset_id);
 }
 
+std::vector<std::string> HDF5File::list(const std::string& group) const
+{
+    // Lambda function pushing the group entries to
+    auto&& list_group = [] (hid_t, const char* name, void* data) -> herr_t
+    {
+        auto& list = *static_cast<std::vector<std::string>*>(data);
+        list.push_back(name);
+        return 0;
+    };
+
+    int idx = 0;
+    std::vector<std::string> result;
+    if (H5Giterate(m_file, group.c_str(), &idx, list_group, &result) < 0) {
```

Comment:
Reading the doc for H5Giterate() at https://docs.hdfgroup.org/hdf5/v1_12/group___h5_g.html#title18 I see that it is deprecated in favour of H5Literate1(), which in turn is deprecated in favour of H5Literate2(). Maybe we should use those?

Also, the newer functions take a parameter for the ordering. It is not clear to me what order will be used with the deprecated H5Giterate(): alphabetic, creation order, or undefined? In any case, the unit test assumes a particular order and I am not sure it can be assumed.

Reply:
this used to be a bit tricky, I used to have to related to 1.6 which didn't have H5Literate. but i guess it's okay now. 

ubuntu 20.04 and rhel7 offers 1.8.x, ubuntu 22.04 and debian 11 offer 1.10.x and future distros are likely to offer 1.12.x or possibly even 1.15. But it seems H5Literate_by_name is available in 1.8, and is a macro in newer releases calling onto H5Literate_by_name1/2 in later releases, so I guess I can use it.

H5Giterate is always in lex order so the test is safe as such, but now it's obviously better since it's explicit.